components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/MessageStoreResource.java

```diff
@@ -0,0 +1,179 @@
+    }
+
+    /**
+     * Sets the list of all available message stores to the response as json
+     * @param messageContext synapse message context
+     **/
+    private void populateMessageStoreList(MessageContext messageContext) {
+        org.apache.axis2.context.MessageContext axis2MessageContext =
+                ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+        SynapseConfiguration synapseConfiguration = messageContext.getConfiguration();
+        Map<String, MessageStore> storeMap = synapseConfiguration.getMessageStores();
+        JSONObject jsonBody = Utils.createJSONList(storeMap.size());
+        storeMap.forEach((key, value) ->
+                addToJsonList(jsonBody.getJSONArray(Constants.LIST), value));
+        Utils.setJsonPayLoad(axis2MessageContext, jsonBody);
+    }
+
+    /**
+     * Sets the information of the specified message store to the response as json
+     * @param messageContext synapse message context
+     * @param name name of the message store
+     * */
+    private void populateMessageStoreData(MessageContext messageContext, String name) {
+        org.apache.axis2.context.MessageContext axis2MessageContext =
+                ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+        SynapseConfiguration synapseConfiguration = messageContext.getConfiguration();
+        MessageStore messageStore = synapseConfiguration.getMessageStore(name);
+        if (Objects.nonNull(messageStore)) {
+            Utils.setJsonPayLoad(axis2MessageContext, getMessageStoreAsJson(messageStore));
+        } else {
+            LOG.warn("Message store " + name + " does not exist");
+            Utils.setJsonPayLoad(axis2MessageContext, Utils.createJsonErrorObject("Message store does not exist"));
+        }
+    }
+
+    /**
+     * Adds the provided message store to the json array
+     * @param list json array
+     * @param messageStore message store
+     * */
+    private void addToJsonList(JSONArray list, MessageStore messageStore) {
+        JSONObject messageStoreObject = new JSONObject();
+        messageStoreObject.put(Constants.NAME, messageStore.getName());
+        messageStoreObject.put(STORE_TYPE_PROPERTY, getStoreType(messageStore));
+        list.put(messageStoreObject);
+    }
+
+    /**
+     * Returns the type of the message store
+     * @param messageStore message store
+     * @return String message store type
+     * */
+    private String getStoreType(MessageStore messageStore) {
+        if (messageStore instanceof JDBCMessageStore) {
+            return JDBCMESSAGESTORE_TYPE;
+        }
+        if (messageStore instanceof JmsStore) {
+            return JMSSTORE_TYPE;
+        }
+        if (messageStore instanceof InMemoryStore) {
+            return INMEMORYSTORE_TYPE;
+        }
+        if (messageStore instanceof RabbitMQStore) {
+            return RABBITMQSTORE_TYPE;
+        }
+        if (messageStore instanceof ResequenceMessageStore) {
```

Comment:
Since ResequenceMessageStore extends JDBCMessageStore, this will always hit the first if statement https://github.com/wso2/micro-integrator/pull/268/files#diff-66e61cbff75759d06e7f60c8c7f043f5R145 and this always be false. We need to move this if check to top I suppose. 
Also a switch or if else ladder would make the code more readable , WDYT ?

Reply:
@sachithKay Shall we fix this?

Reply:
this is fixed now